src/Setup/DbUtil.php

```diff
@@ -30,7 +30,7 @@ public static function encodeDsn($db) {
     return sprintf('mysql://%s:%s@%s/%s',
       $db['username'],
       $db['password'],
-      self::encodeHostPort($db['host'], $db['port']),
+      $db['server']?: self::encodeHostPort($db['host'], $db['port']),
```

Comment:
Good catch -- the `Model::$db` property is documented as using the `server` property, but this function was looking for `host` and `port`. As far as I can see, there's no live/pre-existing code which calls `encodeDsn()`, so I'd vote to make the function simpler and more consistent -- i.e just read `$db['server']` and disregard the possibility of `$db['host']` or `$db['port']`.